### ISSUE TITLE
PHP extension dir isn't properly detected

### ISSUE THREAD

**Issue by eduardomozart** [phpmanager/phpmanager#55]
I'd installed the PHP 7.4 manually at the ``C:\Program Files\PHP\v7.4`` following the steps available at Microsoft Docs: https://learn.microsoft.com/en-us/iis/application-frameworks/scenario-build-a-php-website-on-iis/configuring-step-1-install-iis-and-php

I'd installed PHPManager and renamed the ``php.ini-production`` to ``php.ini`` for PHPManager properly detects the ``php.ini`` file.

When I open the "Enable or disable an extensions" page to enable or enable extensions, it allows me to enable or disable extensions and edit the ``php.ini`` file to reflect those changes, but when I try to load my application, the PHP error logs shows the following warning:

```
PHP Warning: PHP Startup: Unable to load dynamic library 'pdo_sqlite.dll' (tried: C:\php\ext\pdo_sqlite.dll (N├úo foi poss├¡vel encontrar o m├│dulo especificado), C:\php\ext\php_pdo_sqlite.dll (N├úo foi poss├¡vel encontrar o m├│dulo especificado)) in Unknown on line 0
```

It seems already to be reported at the PHP community, but I don't know if the user opened the error report as he/she said it would: https://bugs.php.net/bug.php?id=80416&edit=3

I had to manually edit the ``php.ini`` file and update the ``extension_dir`` directive to point it to my PHP installation dir:

```
extension_dir = C:\Program Files\PHP\v7.4\ext
```

It seems that some validation of ``extension_dir`` is made by the program when the "Recommended INI settings" are applied (function "ValidateExtensionDir" of https://github.com/phpmanager/phpmanager/blob/master/Server/Config/PHPConfigHelper.cs file), but it would be great if it's applied when the PHPManager is setup instead of only when the recommended settings are applied, because it seems that ``extension_dir`` is not set and for some reason it isn't using the exe path for PHP at my IIS instance.

Deploying the PHP to use the default ext path (e.g. ``C:\PHP\ext``) doesn't seems to be the right approach to me because PHPManager support multiple PHP versions that must be deployed at different path locations.

**Comment by lextm** [comment]
There are several things not supported here.

The Microsoft article leads users to install PHP via Web Platform Installer. I never know who produced the MSI installer for PHP, and why PHP needs to be installed to `Program Files`, but that usually leads to more trouble. Web Platform Installer is being deprecated in just a few more days, along with all those MSI packages. So, no plan to support anyone who uses this approach to install PHP.

If, however, you download PHP from https://windows.php.net/download/ and manually configure it on IIS with PHP Manager, I think `extension_dir` will be set to the value you wanted.

**Comment by eduardomozart** [comment]
Hello lextm,
I deployed it to the ``Program Files`` following the MSI logic, but I deployed it manually instead of using the MSI installer. I can redeploy it at ``C:\PHP``, no problems. The problem is that the PHPManager didn't applied the change automatically at my environment, I had to click on "View recommendations" link and apply it manually. The "extension_dir" wasn't set at the ``php.ini`` file and consequently it was falling back to the ``C:\PHP\ext`` directory.
Moving my PHP installation to the fallback would solve the problem without user intervention, but it would create another one: PHPManager and IIS allows you to select a different PHP version, but they must obligatory (and obviously) be installed at different folders (e.g. ``C:\PHP\v7.4`` and ``C:\PHP\v5.4``), so the fallback approach would not work.
Another question I would like to ask: how PHPManager deal with different ``php.ini`` settings for different App Pools? E.g. I create a specific AppPool for GLPI and I would like the extensions and settings applies only for this Site/AppPool. But for what I've seen, it edits the ``php.ini`` files directly. If I create another Site/AppPool and point it to my PHP installation, it will not use the same ``php.ini`` file, loading the same settings and extensions from my GLPI instance, and any change to this file (manually or through PHPManager) would not change both instances? I'm not sure how PHPManager works about it, but I believe I would need to have install the same PHP at different file paths (e.g. ``C:\PHP\glpi\`` and ``C:\PHP\phpmyadmin\``) to have different ``php.ini`` files for each instance, but I would have to update two PHP instances instead of having only one with different ``php.ini`` files for each.
